Music is not working on ubuntu

### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

The issue is created based on a feedback from a user.

No music during application run. Please note that sound effects are OK.
This has started since: 0.9.18

OS: Ubuntu 20.04.3 LTS
fheroes2 engine, version: 0.9.18 - 0.9.20

$ ./fheroes2
fheroes2 engine, version: 0.9.20
15.10.2022 09:41:06: [ERROR] createMusic:  Failed to create a music track from memory. The error: No SoundFonts have been requested
15.10.2022 09:41:18: [ERROR] createMusic:  Failed to create a music track from memory. The error: No SoundFonts have been requested
15.10.2022 09:41:19: [ERROR] createMusic:  Failed to create a music track from memory. The error: No SoundFonts have been requested

The issue is actual for both MIDI and External music setting.

Version 0.9.17 is the last one where I dont get this problem.
No music at all but in game SFX work OK.

### Save file

None.

### Additional info

_No response_

--- a/src/fheroes2/audio/audio_manager.cpp
+++ b/src/fheroes2/audio/audio_manager.cpp
@@ -772,7 +772,11 @@ namespace AudioManager
             // Set the volume for all channels to 0. This is required to avoid random volume spikes at the beginning of the game.
             Mixer::setVolume( -1, 0 );
 
-            Music::SetMidiSoundFonts( midiSoundFonts );
+            // Some platforms (e.g. Linux) may have their own predefined soundfonts, don't overwrite them if we don't have our own
+            if ( !midiSoundFonts.empty() ) {
+                Music::SetMidiSoundFonts( midiSoundFonts );
+            }
+
             Music::setVolume( 100 * Settings::Get().MusicVolume() / 10 );
             Music::SetFadeInMs( 900 );
         }

